rust/src/elem_restriction.rs

```diff
@@ -301,7 +306,7 @@ impl ElemRestriction {
     /// assert_eq!(c, compstride, "Incorrect component stride");
     /// ```
     pub fn comp_stride(&self) -> usize {
-        let mut compstride = 0;
+        let mut compstride: i32 = 0;
```

Comment:
I went ahead and explicitly declared these as i32 since the compiler didn't catch that previous issue - I wonder if the unsafe block is interfering with the type checking?

Reply:
Why??

Reply:
Just let the compiler do its job

Reply:
Because the compiler failed to do its job above, as you noted...

Reply:
I wrote: "I think the compiler sees 1) through the type inference", meaning that the compiler does what he is supposed to do...

Reply:
Right, but the `as usize` should have been a compilation failure because you can't typecast like that in a return, making me think the compiler is making some sort of issue there. It's a short chunk of code, so my suspicion is that the unsafe block is interfering with the compiler in some way.

Reply:
Do you have a reference for this `as` cannot be used on return values? I can't find this anywhere.

Reply:
My reference would be the compiler. I tried it at first and got an error.

Reply:
Huh, it is not erroring on a simple example on the Rust playground, so :shrug:

Reply:
Well, if the reference is the compiler, the compiler didn't throw an error...
The way I understand it is `compstride` has type `i32` inferred from `bind_ceed::CeedElemRestrictionGetCompStride`, and then you cast an `i32` to a `usize`.
What should have gone wrong according to you?

Reply:
The error said something roughly equivalent to 'can't use 'as' in a return statement'
I have zero clues as to where the error went and why it's not showing now

Reply:
I'm not particularly interested in figuring out what happened, so I just removed all of the `: type` stuff. It's probably fine.

Reply:
I'm trying to understand why you say that you suspect the compiler to do something wrong. Is it because you previously had an error than you didn't understand?